src/app/modules/form/form.service.ts

```diff
@@ -151,3 +163,14 @@ export const checkFormSubmissionLimitAndDeactivateForm = async (
     return ok(true)
   }
 }
+
+export const getFormModelByResponseMode = (
+  form: IFormSchema | IPopulatedForm,
```

Comment:
Principle of Least Power - just accept `responseMode` as a parameter?

Reply:
Good point, will fix